src/plots/cartesian/axis_defaults.js

```diff
@@ -75,6 +75,12 @@ module.exports = function handleAxisDefaults(containerIn, containerOut, coerce,
     coerce('range');
     containerOut.cleanRange();
 
+    coerce2('showspikes');
```

Comment:
I see, thanks for writing down the details @rpaskowitz. Your solution indeed _just works_, but it creates a (small) problem: it throws off `Plotly.validate`, which relies on `Plots.supplyDefaults` to determine if user input is invalid or uselessly defined (see test [suite](https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/validate_test.js) for more details). For example, something like:

```js
Plotly.validate(/* */, {
  scene: {
    xaxis: {
        splkedash: 'dash'
    }
  }
}))
```

will be considered valid even though `spikedash` isn't  supported in 3D axes (yet).

-----


So, I'd recommend moving this block to just above this [line](https://github.com/plotly/plotly.js/blob/6e547ef794093c206998103f3065d5f2ab2a20b4/src/plots/cartesian/layout_defaults.js#L184) where it won't be called on 3D axes.